### PR TITLE
Tags Page with more details

### PR DESCRIPTION
**Pull request by panoramix360**
This improves the Tags Page to have much more details on its table #396.

image

I moved some common methods between Stats <-> Tag to do the sorting on the Repo module, maybe we could create a new module for it, but let's see what you guys think.

# Things missing
- [x] Add tests for the new stuff
- Sorting the virtual columns https://github.com/dwyl/mvp/issues/436

### CODE REVIEW

**Comment by nelsonic** [comment]
Dedicated issue for sorting/filtering: https://github.com/dwyl/mvp/issues/436

**Comment by panoramix360** [comment]
hey @nelsonic and @LuchoTurtle!

Finally, I was able to test the files!

Can you guys review it?

Can you help me with this?
- I'm having trouble testing the tags_live, it seems that the creation of tags/items is not producing results on the tests, this is strange since the other live tests work fine. When I add tags to the DB it seems that it doesn't get added for the tests to work.
- I fixed the merge conflicts but I saw that you guys created some modifications regarding lists on the tags controller/view and this might need some work to the index live view of the tags.
- I notice that the test files are being ignored by the formatter on `.formatter.exs`, is this the intended behavior?

I'll wait for your answers!

Thank you and sorry to take a while to create tests for these.

**Comment by panoramix360** [comment]
@nelsonic 

Forgot to mention that I saw that you commented some asserts on the stats_live tests and I fixed them! :D

**Comment by nelsonic** [comment]
@panoramix360 code looks good. ✅ 
Ran it on my `localhost` and looks good. 👌 

image

Added some style to the `Create Tag" Button:

image

Coverage at `98.5%` should be relatively easy to get back up to `100`. 💭 
```
----------------
COV    FILE                                        LINES RELEVANT   MISSED
100.0% lib/api/item.ex                               218       56        0
100.0% lib/api/tag.ex                                101       24        0
100.0% lib/api/timer.ex                              152       40        0
100.0% lib/app/cid.ex                                 25        7        0
100.0% lib/app/color.ex                               90        1        0
100.0% lib/app/date_time_helper.ex                    18        5        0
100.0% lib/app/item.ex                               414       61        0
100.0% lib/app/item_tag.ex                            12        1        0
100.0% lib/app/list.ex                               272       46        0
100.0% lib/app/person.ex                               6        1        0
100.0% lib/app/stats.ex                               91       14        0
100.0% lib/app/tag.ex                                170       31        0
100.0% lib/app/timer.ex                              481       90        0
100.0% lib/app_web/controllers/auth_controller.       25        4        0
100.0% lib/app_web/controllers/init_controller.       44        8        0
100.0% lib/app_web/controllers/list_controller.       93       30        0
100.0% lib/app_web/controllers/tag_controller.e       70       22        0
 98.7% lib/app_web/live/app_live.ex                  569      161        2
100.0% lib/app_web/live/components/nav.ex              7        1        0
100.0% lib/app_web/live/components/table_compon       11        1        0
100.0% lib/app_web/live/stats_live.ex                117       31        0
 46.1% lib/app_web/live/tags_live.ex                  60       13        7
100.0% lib/app_web/router.ex                          51       10        0
100.0% lib/app_web/views/error_view.ex                59       12        0
100.0% lib/app_web/views/list_view.ex                 25        4        0
100.0% lib/app_web/views/nav_view.ex                  16        3        0
100.0% lib/app_web/views/table_component_view.e       19        3        0
  0.0% lib/app_web/views/tag_view.ex                   8        1        1
[TOTAL]  98.5%
----------------
```

Assign to me if you want me to work on the tests. 👌

**Review by nelsonic — DISMISSED** [review]
@panoramix360 Looks great. 👌

File: test/app_web/live/tags_live_test.exs
Comment (panoramix360) [inline test/app_web/live/tags_live_test.exs:27]: @nelsonic can you help me with this test?

I don't know why, but the add_tag is not adding the tags before the asserts below.

So it seems the rendered pages use the default tags created for the test database (shopping, breakfast, etc)

Can you take a look?

**Comment by panoramix360** [comment]
great @nelsonic! 😄 

Can you help me test the tags_live? I just pinged you on a test case there.

Thank you!

I don't why, but

**Comment by panoramix360** [comment]
hey! just noticed but my last message got cut haha

@nelsonic did you have the time to take a look at the tests?

I just returned from vacation and will probably try to fix them again. They are on the `tags_live_test.exs` file.

**Comment by panoramix360** [comment]
@nelsonic @LuchoTurtle 

Happy New Year guys! :D

I just finished this PR with the tests for the tag_live finally! Sorry for this delay, the end of the year was a little troublesome.

Ready to work on another issue!

File: test/app_web/live/tags_live_test.exs
Comment (panoramix360) [reply test/app_web/live/tags_live_test.exs]: done!

File: lib/app/repo.ex
Comment (LuchoTurtle) [inline lib/app/repo.ex:24]: From what I see, `Repo.validate_order/1` is used in `person_with_item_and_timer_count/2` and `llist_person_tags_complete/2` to validate if the sorting order is correct. 

I 100% agree that we should write this function so it can be used anywhere in the project. However, I'm wondering if it's necessary to have this function at all. 

Aren't the aforementioned functions invoked programmatically? What are the scenarios where the sorting order that is passed as arguments to these functions is other than `:asc` and `:desc`? From what I see, this is only possible if the user intentionally messes with the Javascript in the `table_component.html.heex`. If that's the case, [I'd let it crash](https://tinytechtuts.com/2021-let-it-crash-explained/).

**Review by LuchoTurtle — CHANGES_REQUESTED** [review]
Other than my comment, I think this PR is just plain **awesome** 🥳.
I think the only thing missing is a simple write-up in the `BUILDIT.md` detailing the changes you've made in this PR. After that, we'll merge it instantly!

File: lib/app/repo.ex
Comment (panoramix360) [reply lib/app/repo.ex:24]: I think you might be up t o something here.

I agree, I'll remove this method and just leave the code as-is.

Comment (panoramix360) [reply lib/app/repo.ex]: Removed, thank you for the review!

I'll write the BUILDIT.md so this can be ready

Comment (LuchoTurtle) [reply lib/app/repo.ex]: Thank you @panoramix360 ! Once it's ready, you can ping me and I'll approve it and get it merged :)

**Comment by panoramix360** [comment]
hey @nelsonic and @LuchoTurtle 

I just created the new section on the docs with these modifications, but I'm unable to push the branch on this [repo](https://github.com/dwyl/book).

Should I fork it to add a pull request there? I think that I had the permissions earlier for this.

Please, let me know what I need to do, thanks!

**Comment by LuchoTurtle** [comment]
@panoramix360 🤔 you should be able to at least create a PR. If you can't, maybe forking and creating a PR will be faster. 
If that's too much work, you can simply add the section you've written to this repo's `BUILDIT.md`, which also works :)

**Comment by panoramix360** [comment]
great!

it's here!

https://github.com/dwyl/book/pull/100

**Comment by LuchoTurtle** [comment]
Thank you so much @panoramix360 !
Can you copy the text from https://github.com/dwyl/book/pull/100/files you've just created into the `BUILDIT.md` in this PR? Just for completeness :)
I'll approve it after you commit it.

Last stretch! 🏃 Thank you so, so much, again! 🙏

**Comment by panoramix360** [comment]
@LuchoTurtle, done!

**Review by nelsonic — APPROVED** [review]
Excellent Additions @panoramix360 thanks again. 🙏

**Comment by nelsonic** [comment]
"Merging can be performed automatically once the requested changes are addressed." 

image

🤷‍♂️

**Comment by nelsonic** [comment]
I think it's because @LuchoTurtle requested the changes so it's "pending reviewer" ... 

image

**Comment by nelsonic** [comment]
I'm going to manual override as @LuchoTurtle is busy. 👌 

image

**Pull request merged**
